File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestListUUIDValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.test.TypeTester;
import com.dslplatform.ocd.values.ListUUIDInValue.ListUUIDValue;
import com.dslplatform.patterns.ServiceLocator;
import java.util.*;
import org.junit.*;
import static org.junit.Assert.*;

public class TestListUUIDValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
        locator = Bootstrap.init(TestListUUIDValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertTrue(TypeTester.testField(ListUUIDValue.class, "listUUID")
                .resultEquals(List.class, UUID.class));
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testGetter(ListUUIDValue.class, "getListUUID")
                .resultEquals(List.class, UUID.class));
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testSetter(ListUUIDValue.class, "setListUUID", List.class, UUID.class)
                .resultEquals(ListUUIDValue.class));
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestListUUIDValue.scala
package com.dslplatform.ocd.values

import ListUUIDInValue.ListUUIDValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestListUUIDValue extends SpecExtensions {
  def `Test property field type: IndexedSeq[java.util.UUID]`() =
    checkType(
      typeOf[IndexedSeq[java.util.UUID]]
    , typeOf[ListUUIDValue].member("listUUID": TermName).asMethod.returnType
    )

  def `Test property default value: IndexedSeq.empty[java.util.UUID]`() {
    assert(ListUUIDValue().listUUID === IndexedSeq.empty[java.util.UUID])
  }
}
